packages/tailwindcss/src/utils/decode-arbitrary-value.test.ts

```diff
@@ -55,8 +66,8 @@ describe('adds spaces around math operators', () => {
     ['calc(24px+(-1rem))', 'calc(24px + (-1rem))'],
     ['calc(24px_+_(-1rem))', 'calc(24px + (-1rem))'],
     [
-      'calc(var(--10-10px,calc(-20px-(-30px--40px)-50px)',
-      'calc(var(--10-10px,calc(-20px - (-30px - -40px) - 50px)',
+      'calc(var(--10-10px,calc(-20px-(-30px--40px)-50px)))',
+      'calc(var(--10-10px,calc(-20px - (-30px - -40px) - 50px)))',
```

Comment:
This test case didn't have the proper number of closing quotes. I remember that @RobinMalfait once mentioned that we do validate this already at the parser level.

Reply:
I think we only validate the parens in your CSS file (and throw), but we don't validate if it happens in a candidate.

### Input:

```html
<div class="[--foo:calc(var(--10-10px,calc(-20px-(-30px--40px)-50px)]"></div>
```

### Output:

```css
.\[--foo\:calc\(var\(--10-10px\,calc\(-20px-\(-30px--40px\)-50px\)\] {
  --foo: calc(var(--10-10px,calc(-20px - (-30px - -40px) - 50px);
}
```

Reply:
I see, lol!

With this change the closing parentheses would be automatically inserted actually (because we can't encode the absence of them in the `ValueParser`. But I think it would be better if we throw away candidates like this 🤔